modules/core/src/main/java/org/apache/ignite/internal/visor/tx/VisorTxTask.java

```diff
@@ -120,7 +122,11 @@
 
     /** {@inheritDoc} */
     @Nullable @Override protected Map<ClusterNode, VisorTxTaskResult> reduce0(List<ComputeJobResult> results) throws IgniteException {
-        Map<ClusterNode, VisorTxTaskResult> mapRes = new TreeMap<>();
+        int limit = taskArg.getLimit() == null ? DEFAULT_REDUCE_LIMIT : taskArg.getLimit();
+
+        if (limit <= 0) return Collections.emptyMap();
+
+        Map<ClusterNode, VisorTxTaskResult> mapRes = new HashMap<>();
```

Comment:
You changed the data structure here. If there is no good reason for this I strongly advise to leave it as it was.

Reply:
Real mistake, sorry, reverted